client-hc/src/test/java/com/graphhopper/api/GraphHopperWebTest.java

```diff
@@ -0,0 +1,75 @@
+package com.graphhopper.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.graphhopper.GHRequest;
+import com.graphhopper.PathWrapper;
+import com.graphhopper.jackson.Jackson;
+import com.graphhopper.jackson.PathWrapperDeserializer;
+import com.graphhopper.util.shapes.GHPoint;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@RunWith(Parameterized.class)
+public class GraphHopperWebTest {
+
+    private final GraphHopperWeb gh;
+
+    public GraphHopperWebTest(boolean usePost, int maxUnzippedLength) {
+        gh = new GraphHopperWeb(null).setPostRequest(usePost).setMaxUnzippedLength(maxUnzippedLength);
+    }
+
+    /**
+     * Runs the same test with each of the supported traversal modes
+     */
+    @Parameterized.Parameters(name = "POST: {0}, maxUnzippedLength: {1}")
+    public static Collection<Object[]> configs() {
+        return Arrays.asList(new Object[][]{
+                {false, -1},
+                {true, 0},
+                {true, 1000}
+        });
+    }
```

Comment:
This does not run agains graphhopper.com? do we need the parameterization here?

Reply:
Ok, we would need it for GET and POST but the maxUnzippedLength is indeed misleading and not required.

Reply:
(to test that it is applied to OkHttp for POST and GET)

Reply:
One way to do it with Junit5 would be

```java
@ParameterizedTest(name = "POST={0}")
@BooleanSource(values = {true, false})
public void myTest(boolean usePost) {
}
```

Reply:
Thanks, will use this as this is perfect here as I have to create it only once :)

Reply:
Btw: there is no BooleanSource ... it is done via `@ValueSource(booleans = {true, false})` it seems.

Reply:
oh sorry my bad